tests/various/peepopt.ys

```diff
@@ -0,0 +1,63 @@
+read_verilog <<EOT
+module peepopt_shiftmul_0 #(parameter N=3, parameter W=3) (input [N*W-1:0] i, input [$clog2(N)-1:0] s, output [W-1:0] o);
+assign o = i[s*W+:W];
+endmodule
+EOT
+
+prep -nokeepdc
+equiv_opt peepopt
+design -load postopt
+clean
+select -assert-count 1 t:$shiftx
+select -assert-count 0 t:$shiftx t:* %D
+
+####################
+
+design -reset
+read_verilog <<EOT
+module peepopt_shiftmul_1 (output [7:0] y, input [2:0] w);
+assign y = 1'b1 >> (w * (3'b110));
+endmodule
+EOT
+
+prep -nokeepdc
+equiv_opt peepopt
+design -load postopt
+clean
+select -assert-count 1 t:$shr
+select -assert-count 1 t:$mul
+select -assert-count 0 t:$shr t:$mul %% t:* %D
```

Comment:
@cliffordwolf can you check that `peepopt` rightfully cannot optimise this `$shr`?
I wasn't satisfied with the existing tests being in `tests/simple` (which just checks functionality through sim) and moved them to `tests/various` which uses `equiv_opt` as well as `select -assert-count` to ensure that things really are optimised away...

Reply:
@cliffordwolf This remains the sticking point. In the above test and in the new one you gave me that I added in a2eee9eb, it seems that no peep optimisation is being done -- since the design wasn't changed, of course it will remain functionally equivalent. However, we have a multiplier before, and a multiplier after `peepopt`. 
This means that:
1. Either no peepopt is possible with these testcases
2. peepopt is broken
I think it would be more efficient for you to look at at this, please.

Reply:
I've double checked `peepopt_shiftmul_1` and it's intended that this is not optimized. There are actually two reasons for not optimizing this case: (1) the output is larger than the constant factor, so padding the 3-bit chunks to 4 bit would cause incorrect result (at least the way we do padding) and (2) the multiply output has a small bit-width so there may be overflows that would not be handled correctly by our replacement circuit.

(1) is checked by `(GetSize(port(shift, \Y)) > const_factor)` and (2) is checked by `(GetSize(shamt) < factor_bits+GetSize(mul_din))`.

Reply:
Thank you. I shall go ahead and merge this, unless you want to write a test that is eligible to be optimised.